python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -283,17 +348,25 @@ def get_bin_pack_residual(
 
     # A most naive bin packing algorithm.
     nodes = copy.deepcopy(node_resources)
+    used = []
     for demand in resource_demands:
         found = False
-        for node in nodes:
+        node = None
+        i = -1
+        for i in range(len(nodes)):
+            node = nodes[i]
             if _fits(node, demand):
-                _inplace_subtract(node, demand)
                 found = True
+                if strict_spread:
+                    used.append(node)
+                    del nodes[i]
```

Comment:
Hmm is it safe?

Reply:
yes?

Reply:
I am asking because you are removing the entry in place. For example, this will cause index error.

```
In [1]: l = [i for i in range(10)]                                                                                                      

In [2]: for j in range(len(l)): 
   ...:     print(j) 
   ...:     del l[j] 
```

Reply:
+1.  nodes is a list, I think this will shift the list.

Reply:
The goal here is to remove the element (and thus shift the list). 

@rkooo567 your example breaks because you're attempting to use the same iterator over the list _after_ removing an element. This can't happen because we immediately break out of the loop after.